### DISCUSSION TITLE
About same packages in different venvs...

### DISCUSSION THREAD

**Discussion by InternetRebelAKAQuachHungPhong** [bulletmark/pipxu#14] [General]
So, I just realized, when installing a single package via `pipxu` causes a single venv to take around 6 GB, it's be it. Now just a question, do you guys think install another package as/with a launchable command line causes the same dependencies installed in another venv to..symlink to that venv or something, or I just have to create a single venv then inject some launchable packages, also all with `pipxu`?

**Comment by bulletmark** [comment]
Converted to discussion because this is not an issue.

The size of the venv depends completely on which application you installed. E.g. a big package with lots of large dependencies will result in a large venv (and will be the same if you used similar tools like `pipx` or `uv tool`). The point of these tools is to completely isolate the app and it's dependencies from your system and from each other so sharing a venv would defeat the point of that!

`pipxu` does not allow you to install executables when you inject. I think `pipx` can do that so you may consider using that instead.

**Comment by bulletmark** [comment]
BTW, I have 22 packages installed by `pipxu` and the largest venv is 60 MB. That one you are quoting is exceptional, which is it?

**Reply by InternetRebelAKAQuachHungPhong** [reply]
Well, some package related to Streamlit and Torch. It does some captions. And you absolutely could find it on GitHub, but not PyPi.

I see.

**Reply by bulletmark** [reply]
What is the package name?

**Reply by InternetRebelAKAQuachHungPhong** [reply]
S_b_ai.

**Reply by bulletmark** [reply]
I can't find a package or repository of that name. Can you please provide a link?

**Reply by InternetRebelAKAQuachHungPhong** [reply]
https://github.com/abdeladim-s/subsai
